api/discovery/inspector.go

```diff
@@ -0,0 +1,95 @@
+package discovery
+
+import (
+	"context"
+
+	"github.com/dogmatiq/configkit"
+	"github.com/dogmatiq/dodeca/logging"
+	"github.com/dogmatiq/linger/backoff"
+)
+
+// Inspector queries connected clients and notifies an application observer
+// of the applications it has.
+type Inspector struct {
+	// Observer is notified when an application is discovered via a config API
+	// client. It must not be nil.
+	Observer ApplicationObserver
+
+	// Ignore is a predicate function that returns true if the given application
+	// should be ignored.
+	Ignore func(configkit.Application) bool
+
+	// BackoffStrategy controls how long to wait between inspection retries.
+	BackoffStrategy backoff.Strategy
+
+	// Logger is the target for log messages about inspection failures.
```

Comment:
Is it
```suggestion
	// Logger is the target for the log messages about inspection failures.
```
or 
```suggestion
	// Logger is the target for logging the messages about inspection failures.
```
?

Reply:
Or neither? 😄

Reply:
по-моему:

> // Logger is the target for log messages about inspection failures.

This let's you know where any logic messages that might happen to exist (что-нибудь?) will go, and as such lets you know that there are indeed log messages.

> // Logger is the target for **the** log messages about inspection failures.

This is valid, but to me sounds like the **user** has provided some specific (что-то) log messages already.

> // Logger is the target for logging the messages about inspection failures.

This would work too, but *almost* sounds like you're talking about logging Dogma messages, rather than "log messages" as a noun-phrase, or compound noun or whatever it's called.

Reply:
thanks for the elaborate explanation!

Reply:
I *think* you feel how I feel when I ask you to explain some difference between two equivalent Russian sentences :D